src/event/metric.rs

```diff
@@ -289,28 +290,26 @@ impl Metric {
 
     /// Convert the metrics_runtime::Measurement value plus the name and
     /// labels from a Key into our internal Metric format.
-    pub fn from_metric_kv(key: &metrics::Key, handle: &metrics_util::Handle) -> Self {
+    pub fn from_metric_kv(key: &metrics::Key, handle: &Handle) -> Self {
         let value = match handle {
-            metrics_util::Handle::Counter(_) => MetricValue::Counter {
-                value: handle.read_counter() as f64,
+            Handle::Counter(counter) => MetricValue::Counter {
+                value: counter.count() as f64,
             },
-            metrics_util::Handle::Gauge(_) => MetricValue::Gauge {
-                value: handle.read_gauge() as f64,
+            Handle::Gauge(gauge) => MetricValue::Gauge {
+                value: gauge.gauge(),
             },
-            metrics_util::Handle::Histogram(_) => {
-                let values = handle.read_histogram();
-                // Each sample in the source measurement has an
-                // effective sample rate of 1.
-                let samples = values
+            Handle::Histogram(histogram) => {
+                let mut buckets: Vec<(f64, u32)> = Vec::with_capacity(32);
+                histogram.buckets(&mut buckets);
+                let buckets: Vec<Bucket> = buckets
                     .into_iter()
```

Comment:
This is the only caller of `Histogram::buckets`. Could it implement an iterator instead of pushing into an over-sized `Vec` that is immediately destroyed?

Reply:
Good idea.